### ISSUE TITLE
[Feature]: GFX11 512 Head Dim

### ISSUE THREAD

**Issue by Beinsezii** [ROCm/aotriton#168]
### Suggestion Description

256 is still the max in torch 2.11 despite it being required for most image model VAEs

```py
import torch
from diffusers.models.autoencoders.autoencoder_kl import AutoencoderKL

with (
    torch.inference_mode(),
    torch.nn.attention.sdpa_kernel(
        [  # Disable MATH backend
            torch.nn.attention.SDPBackend.FLASH_ATTENTION,
            torch.nn.attention.SDPBackend.EFFICIENT_ATTENTION,
        ]
    ),
):
    dv = torch.device("cuda:0")
    dt = torch.bfloat16
    vae: AutoencoderKL = AutoencoderKL.from_pretrained("Jiali/stable-diffusion-1.5", subfolder="vae").to(
        dtype=dt, device=dv
    )
    vae.config.force_upcast = False

    vae.decode(torch.rand([1, 4, 128, 128], device=dv, dtype=dt))
```

Seems like it was disabled in torch so assuming its an AOTriton issue?
https://github.com/ROCm/pytorch/blob/3aaa914af1e6fb268b242bfb871e614fbdb6c1bc/aten/src/ATen/native/transformers/cuda/sdp_utils.cpp#L158-L159

### GPU

7900 XTX

**Comment by xinyazhang** [comment]
hdim > 256 is disabled intentionally on all gfx11 due to their limited LDS and register file size, and hence Triton compiler has great trouble generating `BLOCK_DMODEL=512` GPU binary historically: https://github.com/ROCm/aotriton/blob/5b05b446e38d4db3f5c68b2643a2f71d2ed2a8af/v3python/rules/flash/_common.py#L65-L68

**Comment by Beinsezii** [comment]
> hdim > 256 is disabled intentionally on all gfx11 due to their limited LDS and register file size, and hence Triton compiler has great trouble generating `BLOCK_DMODEL=512` GPU binary historically:

Used to be available back in torch ~2.8, ever since max size for image models has been severely gimped since there's zero gfx11 efficient attention implementations with 512D. Is the plan to just not support this anymore then?

**Comment by xinyazhang** [comment]
> Is the plan to just not support this anymore then?

The current challenges to have more extensive coverage of gfx11 are
* Triton is more fragile on gfx11 (yes Triton is more stable and performant on the newer gfx1201)
* It demands more HW resources to run the full tuning (seqlen 8192 + hdim 512)
* As a combination of the two problems above, the tuning will probably never complete due to frequent GPU segfault&restarting

We will revisit this feature after making our tuning process "smarter".
